assembl/lib/antivirus.py

```diff
@@ -0,0 +1,69 @@
+from tempfile import mkdtemp
+from subprocess import call
+from os import unlink, rmdir
+from os.path import join
+from random import random
+
+from pyramid.path import DottedNameResolver
+
+from . import config
+
+
+resolver = DottedNameResolver(__package__)
+
+
+class AntiVirus(object):
+    def check(self, data, suffix=None):
+        return NotImplementedError()
+
+
+class MockPositiveAntiVirus(AntiVirus):
+    "A Mock antivirus for testing. Reports no file infected."
+    def check(self, data, suffix=None):
+        return True
+
+
+class MockRandomAntiVirus(AntiVirus):
+    "A Mock antivirus for testing. Reports 10%% files infected."
+    def check(self, data, suffix=None):
+        return random() > 0.1
+
+
+class FileAntiVirus(AntiVirus):
+    def check(self, data, suffix=None):
```

Comment:
Ugh, so much heavy lifting per file. This is going to crush CPU under moderate load. For first pass, though, it'll do.